### PR TITLE
Scalpel can now cut pizza.

### PR DESCRIPTION
**Pull request by jaw-sh**
## About The Pull Request
Enables scalpel tools to cut pizza. Consolidated a bunch of pizza-specific `MakeProcessable` proc defines that were just copy+pastes of each other.

## Why It's Good For The Game
You sit in medbay with your young doctor, he says he is hungry.  You agree to get him pizza. You call cargo, who brings you the box. Pizza stares at you from the table. You cannot cut it. Your palms, holding a scalpel, get sweaty, and your doctor complains. He says he is hungry. Your mind strains, searching for an answer in a world of spaghetti code. You try and ask the chef for his knife, but he has been gibbed. His knife is gone. Your doctor is hungry, he complains. The cargo delivery man stares at you, his delivery box empty, his acquisition form needing a signature. Your coworker cries tears of a pizza-less doctor. It is raining spaghetti now, you have knife embedded in your left temple and you do not understand. The doctor weeps in the corner, he is starving. Starving for the pizza.

## Changelog
:cl:
fix: Pizza can now be cut with a scalpel.
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
Ask cargo to print a kitchen knife from the autolathe before they deliver the pizza to you.

**Comment by ArcaneMusic** [comment]
Ask medical to make some plastic so you can print a plastic knife to cut it with.

**Comment by jaw-sh** [comment]
who the fuck eats pizza with a plastic knife

**Comment by ArcaneMusic** [comment]
> who the fuck eats pizza with a plastic knife

The same person who doesn't have a metal kitchen knife. 😏

**Comment by Geikkamir** [comment]
> 
> who the fuck eats pizza with a ~~plastic~~ knife

**Comment by Mickyan** [comment]
they did surgery on a pizza

**Comment by interestingusernam3** [comment]
Add this ability to bone saw too, it's pretty much a heavy duty pizza cutter after all

**Comment by TheBonded** [comment]
> Add this ability to bone saw too, it's pretty much a heavy duty pizza cutter after all

pretty sure circular saws can, as well as (maybe) surgical drills

**Comment by CRITAWAKETS** [comment]
can swords cut pizza aswell? i want to use the energy katana to cut my pineapple pizza

actually why not just have it work on sharpness? its not like pizza is immune to things like glass shards

**Comment by Qustinnus** [comment]
I'm not a big fan of using separate elements for this, tool behavior shouldn't be an enum in that case

Also the reason make processable is not using a slice type has a reason that was discussed in the original PR.

**Comment by optimumtact** [comment]
we are in a feature freeze until the end of november dude

**Comment by jaw-sh** [comment]
This was existing functionality that was removed on accident at some point but okay.

**Comment by optimumtact** [comment]
proof?

**Comment by Qustinnus** [comment]
before food refactor cutable food checked is_sharp(), I refactored it to use TOOL_KNIFE

**Comment by Qustinnus** [comment]
and is_sharp was true on basically anything with a pointy end

**Comment by optimumtact** [comment]
ok, I'll take your word for it

**Comment by Qustinnus** [comment]
@jaw-sh 

btw if you want maybe you could fix a small thing I noticed about moldy pizza; the cytology element gets added on MakeProcessable. Could you instead move that to Initialize()?

**Comment by jaw-sh** [comment]
@Qustinnus  While testing this commit, I noticed there was a moldy bread item that was also not able to be swabbed. I've made that swabbable as well.

image

**Comment by Qustinnus** [comment]
thanks king, didn't know that existed.

**Comment by wesoda25** [comment]
What’s wrong with using sharpness? It makes sense.

**Comment by Qustinnus** [comment]
something being sharp does not equal it being a good tool to cut food, and using non-unified systems like sharpness makes it difficult to make systems like the processable element.

**Comment by Tlaltecuhtli** [comment]
shouldnt scalpels just have the knife tool flag? like they  are knifes and should cut other stuff other than pizzas

**Comment by Fikou** [comment]
> shouldnt scalpels just have the knife tool flag? like they are knifes and should cut other stuff other than pizzas

tool flags arent really flags, they are only able to be one thing at a time, some servers like eris have multiple tool behaviours on items and if you click on something with an item that has multiple tool behaviours you select what tool you want to use in a radial menu

**Comment by Qustinnus** [comment]
I've been conceptualizing a tool refactor to account for this but we might as well do it like this for now until we get the proper fix. the processable element is super cheap so theres really no problem

File: code/game/objects/items/food/pizza.dm [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-
+// Pizza (Whole)
 /obj/item/food/pizza
 	icon = 'icons/obj/food/pizzaspaghetti.dmi'
 	w_class = WEIGHT_CLASS_NORMAL
 	max_volume = 80
 	food_reagents = list(/datum/reagent/consumable/nutriment = 28, /datum/reagent/consumable/nutriment/protein = 3, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
 	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
 	foodtypes = GRAIN | DAIRY | VEGETABLES
+	var/obj/item/food/pizzaslice/slice_type
```
Comment (Cyberboss) [inline code/game/objects/items/food/pizza.dm:9]: DMDoc for this var

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/food/pizza.dm (hunk 1 above)
Comment (jaw-sh) [reply code/game/objects/items/food/pizza.dm:10]: Like this?
```
	/// type is spawned 6 at a time and replaces this pizza when processed by cutting tool
	var/obj/item/food/pizzaslice/slice_type```

Comment (ArcaneMusic) [reply code/game/objects/items/food/pizza.dm:10]: That works.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**